apps/mobile/src/app/account/[account]/activity.tsx

```diff
@@ -1,27 +1,31 @@
-import { useAccountAddresses } from '@/hooks/use-account-addresses';
-import { useAccountActivityQuery } from '@/queries/activity/account-activity.query';
+import { configureAccountParamsSchema } from '@/app/account/[account]/index';
+import { FetchWrapper } from '@/components/loading';
+import { ActivityCell, ActivityEmpty, ActivityLayout, hasActivity } from '@/features/activity';
+import { useAccountActivity } from '@/queries/activity/account-activity.query';
 import { deserializeAccountId } from '@/store/accounts/accounts';
 import { useLocalSearchParams } from 'expo-router';
 
-import { ActivityLayout } from '../../activity/activity.layout';
-import { ActivityCell } from '../../activity/components/activity-cell';
-import { configureAccountParamsSchema } from './index';
-
 export default function AccountActivityScreen() {
   const params = useLocalSearchParams();
   const { accountId, accountName } = configureAccountParamsSchema.parse(params);
   const { fingerprint, accountIndex } = deserializeAccountId(accountId);
-  const accountAddresses = useAccountAddresses(fingerprint, accountIndex);
-  const { data: activity, isLoading } = useAccountActivityQuery(accountAddresses);
+  const activity = useAccountActivity(fingerprint, accountIndex);
 
-  if (isLoading) {
-    return null;
-  }
   return (
     <ActivityLayout accountName={accountName}>
-      {activity?.map((activity, index) => (
-        <ActivityCell key={`activity.${index}`} activity={activity} />
-      ))}
+      <FetchWrapper data={activity}>
+        {hasActivity(activity) ? (
+          activity.state === 'success' &&
+          activity.value?.map((activity, index) => {
```

Comment:
Perhaps better as a `slice(0, 5).map(...)` but since it's getting changed later anyway maybe doesn't matter.

Reply:
Thanks! 

I've actually updated this in my next PR already as @tigranpetrossian recommended using `slice` also:
```
export function ActivityList({ activity, mode = 'full' }: ActivityListProps) {
  const Component = mode === 'widget' ? ActivityCard : ActivityListItem;
  return (
    <Box>
      {activity.slice(0, mode === 'widget' ? 5 : undefined).map((activity, index) => (
        <Component key={`activity.${index}`} activity={activity} />
      ))}
    </Box>
  );
}

```